plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3select/csv/S3SelectCsvRecordReader.java

```diff
@@ -93,6 +96,16 @@ public SelectObjectContentRequest buildSelectObjectRequest(Properties schema, St
         selectObjectOutputSerialization.setCsv(selectObjectCSVOutputSerialization);
         selectObjectRequest.setOutputSerialization(selectObjectOutputSerialization);
 
+        // Scan range requests to S3 Select allow us to enable splits,
+        // by providing a finer granularity than file level.
+        if (CompressionType.NONE.equals(compressionType)) {
```

Comment:
for CSV we should also check if we allow quotedDelimiters or not. Scan range is only supported for CSV without quoted delimiters. (https://docs.aws.amazon.com/AmazonS3/latest/userguide/selecting-content-from-objects.html) It looks like we do not set AllowQuotedDelimiter to true already, but we can still check it on scan-range if in future we have another PR that introduces that change.

Reply:
Updated, please check